### PR TITLE
config: Refactor config loading

### PR DESCRIPTION
**Pull request by kakkoyun**
Add warning messages for falling back to prevent #2202 

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

File: cmd/parca-agent/main.go [hunk 1]
```diff
@@ -555,7 +555,13 @@ func run(logger log.Logger, reg *prometheus.Registry, flags flags) error {
 		if err != nil {
 			return fmt.Errorf("failed to read config: %w", err)
 		}
-		cfg = cfgFile
+		if cfg == nil {
+			level.Warn(logger).Log("msg", "config file is empty, using default config")
+		} else {
+			cfg = cfgFile
```
Comment (javierhonduco) [inline cmd/parca-agent/main.go:561]: Did you mean to move this to the block above?

File: pkg/config/config.go [hunk 2]
```diff
@@ -36,13 +35,17 @@ func (c Config) String() string {
...
+	if len(b) == 0 {
+		return nil, nil
+	}
+
+	// TODO: Sanitize the input labels if it becomes a problem for the user.
+	// https://github.com/prometheus/prometheus/blob/1bfb3ed062e99bd3c74e05d9ff9a7fa4e30bbe21/util/strutil/strconv.go#L51
 	cfg := &Config{}
 
-	dec := yaml.NewDecoder(bytes.NewBufferString(s))
-	dec.KnownFields(true)
```
Comment (javierhonduco) [inline pkg/config/config.go:43]: Why was this removed?

Comment (kakkoyun) [reply pkg/config/config.go:43]: We are using a different API and it doesn't have this method.

**Comment by kakkoyun** [comment]
@javierhonduco should be cleaner now.
